### PR TITLE
Closes #356

### LINKED ISSUE #356: `StoppingMTDCV` class cleanup.
Please edit appropriately below if needed.

To do:

- [x] Revise documentation
  - [x] RStudio outline section dividers (use `# function or class name ----`)
  - [x] Add lifecycle badge (as a default take "stable" for existing functions or classes)
  - [x] Replace `##'` by `#'`
  - [x] Add to `pkgdown.yml`
  - [x] Remove file history and the timestamp in comments on top of file
- [x] Add unit tests
  - [x] For classes:
    - [x] Test default constructor
    - [x] For reference classes: test methods
  - [x] For functions: as usual
- [x] Refactor as needed
  - [x] Code styling (indentation etc.) - install package `styler` and use corresponding add-in
  - [x] Keep externally used names so that we don't break downstream code
  - [x] Remove `keywords` tags
  - [x] For functions or methods:
    - [x] Add assertions for arguments using `checkmate` functions
    - [x] If it is too long, break up into pieces (rule of thumb: maximum 50 lines of code)
    - [x] Consider renaming internal variables etc. if needed
    - [x] Avoid explicit use of `return`
  - [x] For classes:
    - [x] move function definitions outside if possible (e.g. move validation function as `validity_subjects()` to `Data-validity.R` and have corresponding test file `test-Data-validity.R`)
    - [x] In `setClass` function, just use the `slots` instead of outdated `representation`
    - [x] Remove things like `validObject(.Data())`

### PR DESCRIPTION
**Pull request by wwojciech**
Closes #356

### CODE REVIEW

File: R/Rules-validity.R [hunk 1]
```diff
@@ -435,8 +431,12 @@ v_stopping_mtd_cv <- function(object) {
     "target must be probability value from (0, 1) interval"
```
Comment (danielinteractive) [inline R/Rules-validity.R:431]: need to adapt this?
```suggestion
    "target must be probability value from (0, 1] interval"
```

File: tests/testthat/test-Rules-class.R [hunk 2]
```diff
@@ -422,18 +422,25 @@ test_that("StoppingMTDdistribution object can be created with user constructor",
...
-
-test_that("StoppingMTDCV object can be created with user constructor", {
+test_that("StoppingMTDCV object can be created with user constructor (default)", {
   result <- expect_silent(StoppingMTDCV())
   expect_valid(result, "StoppingMTDCV")
+  expect_identical(result@target, 0.3)
+  expect_identical(result@thresh_cv, 40)
+})
+
+test_that("StoppingMTDCV object can be created with user constructor (de", {
```
Comment (danielinteractive) [inline tests/testthat/test-Rules-class.R:439]:
```suggestion
test_that("StoppingMTDCV object can be created with user constructor (default)", {
```
?

File: tests/testthat/test-Rules-validity.R [hunk 3]
```diff
@@ -880,48 +872,47 @@ test_that("v_stopping_mtd_distribution returns message for non-valid prob", {
...
 
-  expect_equal(
-    v_stopping_mtd_cv(object),
-    c(
-      "target must be probability value from (0, 1) interval",
-      "thresh_cv must be percentage > 0"
-    )
-  )
+  # Changing `target` so that it does not represent allowed probability value.
+  object@target <- 1
```
Comment (danielinteractive) [inline tests/testthat/test-Rules-validity.R:885]: I thought that should be ok?

**Review by danielinteractive — COMMENTED** [review]
thanks Wojciech!

File: tests/testthat/test-Rules-class.R (hunk 2 above)
Comment (wwojciech) [reply tests/testthat/test-Rules-class.R:439]: Oh, that is a typo. Tx!

File: tests/testthat/test-Rules-validity.R (hunk 3 above)
Comment (wwojciech) [reply tests/testthat/test-Rules-validity.R:885]: See the comment for the first conversation, above.

File: R/Rules-validity.R [hunk 4]
```diff
@@ -435,8 +431,12 @@ v_stopping_mtd_cv <- function(object) {
...
   )
   v$check(
-    test_probability(object@thresh_cv / 100, bounds_closed = FALSE),
+    test_probability(object@thresh_cv / 100, bounds_closed = c(FALSE, TRUE)),
```
Comment (danielinteractive) [inline R/Rules-validity.R:434]: OK I guess I confused with this here - maybe good if you double check that the tests for this are correct?

Comment (wwojciech) [reply R/Rules-validity.R:434]: I think this is ok, so `tresh_cv = 100` means that we stop for sure, regardless of the posterior dose samples, as:
```
do_stop <- (mtd_cv <= stopping@thresh_cv)
```
and `mtd_cv` takes on value from `[0, 1]`.

**Review by danielinteractive — APPROVED** [review]
thanks I just double checked the tests you have for the `thresh_cv` and looks good

**Pull request merged**